tests/test_data_iterator.py

```diff
@@ -0,0 +1,110 @@
+from re import L
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import pandas.testing as pd_test
+import pytest
+
+from packages.python.accern_data import create_data_client, DT_FORMAT
+from packages.python.accern_data.util import (
+    DEFAULT_CHUNK_SIZE,
+    EXAMPLE_URL,
+    load_json,
+)
+
+
+@pytest.mark.parametrize("chunk_size", [1, 5, 1000, None])
+def test_csv_full_iterator(chunk_size: Optional[int]) -> None:
+    start_date = "2022-01-03"
+    end_date = "2022-03-04"
+    client = create_data_client(EXAMPLE_URL, "SomeRandomToken")
+    client.set_mode("csv", split_dates=False, chunk_size=chunk_size)
+    dataframe = pd.read_csv("./tests/data/data-2022.csv")
+    if chunk_size is not None:
+        n_full_chunks = dataframe.shape[0]//chunk_size
+    df_lengths = []
+    dfs: List[pd.DataFrame] = []
+    for df in client.iterate_range(start_date=start_date, end_date=end_date):
+        df_lengths.append(df.shape[0])
```

Comment:
how do you confirm the lengths?

Reply:
by comparing it with chunk_size. since it's combined csv case, all batches would have no of rows equal to chunk_size (not always true for last batch)

Reply:
but it's not the chunk size
it is only the chunk size if there is enough data